image-classifier/index.ts

```diff
@@ -0,0 +1,165 @@
+    await imageClassifier.setOptions({ runningMode });
+  }
+
+  // imageClassifier.classify() returns a promise which, when resolved, is a ClassificationResult object.
+  // Use the ClassificationResult to print out the results of the prediction.
+  const classificationResult = await imageClassifier.classify(event.target);
+  // Write the predictions to a new paragraph element and add it to the DOM.
+  const classifications = classificationResult.classifications;
+
+  const p = event.target.parentNode.childNodes[3];
+  p.className = 'classification';
+  p.innerText = `Classification: ${classifications[0].categories[0].categoryName}
+                 Confidence: ${Math.round(parseFloat(`${classifications[0].categories[0].score}`) * 100)}%`;
+    // 'Classificaton: ' +
+    // classifications[0].categories[0].categoryName +
+    // '\n Confidence: ' +
+    // Math.round(parseFloat(`${classifications[0].categories[0].score}`) * 100) +
+    // '%';
+}
+
+/********************************************************************
+// Demo 2: Continuously grab image from webcam stream and classify it.
+********************************************************************/
+
+// Check if webcam access is supported.
+function hasGetUserMedia() {
+  return !!(navigator.mediaDevices && navigator.mediaDevices.getUserMedia);
+}
+
+// Get classification from the webcam
+async function predictWebcam() {
+  // Do not classify if imageClassifier hasn't loaded
+  if (imageClassifier === undefined) {
+    return;
+  }
+  // if image mode is initialized, create a new classifier with video runningMode
+  if (runningMode === 'image') {
+    runningMode = 'video';
+    // await runDemo();
+    await imageClassifier.setOptions({ runningMode: runningMode });
+  }
+  let nowInMs = Date.now();
+  // Start classifying the stream.
+  const classificationResult = await imageClassifier.classifyForVideo(
+    video,
+    nowInMs
+  );
+  video.style.height = videoHeight;
+  video.style.width = videoWidth;
+  webcamPredictions.style.width = videoWidth;
+
+  const classifications = classificationResult.classifications;
+  webcamPredictions.className = 'webcamPredictions';
+  webcamPredictions.innerText = `Classification: ${classifications[0].categories[0].categoryName}
+                                 Confidence: ${Math.round(parseFloat(`${classifications[0].categories[0].score}`) * 100)}%`;
+    // 'Classification: ' +
+    // classifications[0].categories[0].categoryName +
+    // '\n Confidence: ' +
+    // Math.round(parseFloat(classifications[0].categories[0].score) * 100) +
+    // '%';
+
+  // Call this function again to keep predicting when the browser is ready.
+  if (webcamRunning === true) {
+    window.requestAnimationFrame(predictWebcam);
+  }
+}
+
+// Enable the live webcam view and start classification.
+async function enableCam(event) {
+  if (imageClassifier === undefined) {
+    return;
+  }
+
+  if (webcamRunning === true) {
+    webcamRunning = false;
```

Comment:
This looks like it stops our classification loop, but not the webcam stream.  Is that intentional?  I think that behavior's kinda nice because then the user can see the webcam and toggle on/off predictions at will, but if that's the case, then we should change our variable name to reflect this.  Like maybe `webcamClassifying` instead of `webcamRunning`?